src/plots/cartesian/set_convert.js

```diff
@@ -608,6 +608,12 @@ module.exports = function setConvert(ax, fullLayout) {
         var rangebreaksIn = ax.rangebreaks || [];
         var bnds, b0, b1, vb, vDate;
 
+        if (!rangebreaksIn._cachedVals) {
+            rangebreaksIn._cachedVals = rangebreaksIn.map(brk => {
+                return brk.values ? Lib.simpleMap(brk.values, ax.d2c).sort(Lib.sorterAsc) : null;
+            });
+        }
+
```

Comment:
Cache only if brk.enabled and cache the brk.patterns here as well and use them below. Also prefer the name `_cachedValues` instead of `_cachedVals`.

```
        if (!rangebreaksIn._cachedPatterns) {
            rangebreaksIn._cachedPatterns = rangebreaksIn.map(brk => {
                return brk.enabled && brk.bounds ? Lib.simpleMap(brk.bounds, brk.pattern ?
                        cleanNumber :
                        ax.d2c // case of pattern: ''
                ) : null;
            });
        }
        if (!rangebreaksIn._cachedValues) {
            rangebreaksIn._cachedValues = rangebreaksIn.map(brk => {
                return brk.enabled && brk.values ? Lib.simpleMap(brk.values, ax.d2c).sort(Lib.sorterAsc) : null;
            });
        }

```